### PR TITLE
Update deleteStatefulSet to compare with desired labels

### PR DESCRIPTION
**Pull request by gkrenn**
deleteStatefulSetIfOldLabelsAreUsed now generally compares if the labels have changed

### CODE REVIEW

File: controllers/capability/reconciler.go [hunk 1]
```diff
@@ -174,15 +170,17 @@ func (r *Reconciler) deleteStatefulSetIfOldLabelsAreUsed(desiredSts *appsv1.Stat
 		return false, err
 	}
 
-	if _, ok := currentSts.Labels[OldKeyActiveGate]; !ok {
-		return false, nil
+	for k := range desiredSts.Labels {
```
Comment (lrgar) [inline controllers/capability/reconciler.go:173]: Alternatively, you can go with `reflect.DeepEqual(currentSts.Labels, desiredSts.Labels)`

File: controllers/capability/reconiler_test.go [hunk 2]
```diff
@@ -216,14 +216,20 @@ func TestReconcile_DeleteStatefulSetIfOldLabelsAreUsed(t *testing.T) {
...
 	desiredSts, err = r.buildDesiredStatefulSet()
 	require.NoError(t, err)
+	correctLabels := make(map[string]string)
+	for key, value := range desiredSts.Labels {
+		correctLabels[key] = value
+	}
+	delete(desiredSts.Labels, KeyActiveGate)
+	delete(desiredSts.Labels, KeyDynatrace)
+	delete(desiredSts.Labels, KeyFeature)
+	desiredSts.Labels["activegate"] = "dynakube"
```
Comment (lrgar) [inline controllers/capability/reconiler_test.go:228]: If the intention is just to replace the whole map, I may suggest doing,

```go
correctLabels := desiredSts.Labels
desiredSts.Labels = map[string]string { "activegate": "dynakube" }
```

**Review by lrgar — APPROVED** [review]
Added some suggestions, but good otherwise.

File: controllers/capability/reconiler_test.go (hunk 2 above)
Comment (gkrenn) [reply controllers/capability/reconiler_test.go]: Yes, that's way better

File: controllers/capability/reconciler.go (hunk 1 above)
Comment (gkrenn) [reply controllers/capability/reconciler.go]: Ok, thx

Did not know that

**Review by lrgar — APPROVED** [review]

**Review by DTMad — APPROVED** [review]

**Pull request merged**